viewing_party/party.py

```diff
@@ -0,0 +1,129 @@
+#wave 1
+
+def create_movie(title, genre, rating):
+    if title ==  None:
+        return None
+    if genre == None:
+        return None
+    if rating == None:
+        return None
+    else:
+        movie = {"title": title, "genre": genre, "rating": rating}
+        return movie
+
+def add_to_watched(user_data, movie):
+    user_data['watched'].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data['watchlist'].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    for movie in user_data['watchlist']:
+        if movie['title'] == title:
+            user_data['watchlist'].remove(movie)
+            user_data["watched"].append(movie)
+        if user_data['watched'] == []:
+            user_data["watched"].append(movie)
+
+    return user_data
+
+#wave 2 pt 1 
+
+def get_watched_avg_rating(user_data): 
+    if len(user_data["watched"]) == 0:
+        return 0 
+    else:
+        ratings = []
+        for movie in user_data['watched']:
+            ratings.append(movie["rating"])
+        average = sum(ratings)/len(ratings)
+        return average 
+
+ # wave 2 pt 2
+ 
+def get_most_watched_genre(user_data):
+    genres_watched = []
+
+    if len (user_data["watched"]) == 0:
+        return None
+
+    for movie in user_data["watched"]:
+        genres_watched.append(movie["genre"])
+
+    freq = {}
+
+    for genre in genres_watched:
+        if not genre in freq:
+            freq.update({genre:1})
+        else:
+            freq[genre] += 1 
+
+    popular_genre = max(freq, key = freq.get)
+
+    return popular_genre
+
+#wave 3 pt 1
+
+def get_unique_watched(user_data):
+    user_watched = user_data["watched"]
+    friends_watched = user_data["friends"]
```

Comment:
I like that you created variables to make it easier to isolate the data you are working with